translation: improve translatable strings

illwieckz opened the pull request:
- remove ending space or new line
- move some color codes out of the string
- remove endline from some log

Also:

- exclude maprotation command log from game pot

slipher commented on src/cgame/cg_tutorial.cpp:
```diff
@@ -441,55 +476,62 @@ static void CG_HumanText( char *text, playerState_t *ps )
 	if ( BG_InventoryContainsUpgrade( UP_JETPACK, ps->stats ) )
 	{
-		Q_strcat( text, MAX_TUTORIAL_TEXT, va( _( "Hold %s to use the %s\n" ),
+		Q_strcat( text, MAX_TUTORIAL_TEXT, va( _( "Hold %s to use the %s" ),
```
We would get better translation if strings like this said `Hold %s to use the jetpack` rather than `Hold %s to use the %s`.

illwieckz replied:
Well, this hardcoding is only doable for two or three things, the jetpack, the medkit, and the armoury, probably no more. If we consider those names not common names but proper nouns (like if we name a branded product), I guess this would work in most languages. Languages that would be badly translated are the ones with declensions (like Latin) because those names would lack the proper declension (here the accusative one), but then only those 2 or 3 sentences among all the tutorial ones would be correct. I would take care of those 2 or 3 strings the day we want to make all strings from tutorial correct in such languages. We may even chose another solution, where declension languages would provide the accusative variation for the word and the function would pick the accusative variation if it exists. So to me it's too early to decide on that.

illwieckz replied:
I finally did what's suggested (hardcoding those names), as there is already some sentences doing that, for example:

```sh
_( "Find an Armoury, Drill or Reactor for more ammo." ) );
_( "Find an Armoury for more ammo." ) );
```

But I would not oppose to doing the absolute opposite and modifying those other sentences instead, so they rely on entity names programmatically.

illwieckz commented on src/cgame/cg_tutorial.cpp:
```diff
@@ -124,7 +124,7 @@ static const char *CG_KeyNameForCommand( const char *command )
 	}
 
-	return "(⚠ BUG)"; // shouldn't happen: if it does, BUG
+	return va( "(⚠  %s )", _("BUG") ); // shouldn't happen: if it does, BUG
```
Ok, I reverted that.

illwieckz commented:
> One thing it maybe time to decide if we want to change that, is that if we want to end those tutorial sentences with a dot.
> 
> In fact we already have two of them, when there are two sentences in the same line:
> 
> ```c++
> va( _( "Press %s to spit. Spits can slow humans and remove fire." ),
> ```
> 
> _Note:_ I just noticed we may even split those translatable sentences, but that changes nothing to the decision about dot-ended sentences.

I added ending dots to sentences, and split that one into two translatable strings.

necessarily-equal reviewed (approved):
Didn't test in game, but LGTM

slipher commented:
> I added ending dots to sentences, and split that one into two translatable strings.

Why split it? More finely chopped strings = less context, hence worse translations.

illwieckz commented:
> > I added ending dots to sentences, and split that one into two translatable strings.
> 
> Why split it? More finely chopped strings = less context, hence worse translations.

OK then, I liked the small sentences. Reverted.

illwieckz commented:
Question about “Spits can slow humans and remove fire.”, is “Spits can slow down humans and remove fire.” better or not?

illwieckz commented:
Once this is merged I generate `game.po` to make it discoverable by Weblate, and I submit Unvanquished to Weblate for evaluation for free open source plan.

slipher commented:
> Question about “`Spits can slow humans and remove fire.`”, is “`Spits can slow down humans and remove fire.`” better or not?

They're both acceptable; for tutorial text I'd keep the former since it should be concise.

`Spits` is wrong though; spit is an uncountable noun. Should be `Split can slow`...

illwieckz commented:
> `Spits` is wrong though; spit is an uncountable noun. Should be `Split can slow`...

Thanks, it's now fixed.

bmorel commented:
lgtm.

slipher commented on src/utils/generate_pot.sh:
```diff
@@ -49,7 +49,10 @@ cd "${repo_dir}"
+	-name '*.cpp' \
+	-a ! -name sg_admin.cpp \
+	-a ! -name sg_cmds.cpp \
+	-a ! -name sg_maprotation.cpp \
```
You have to add this to the xgettext command at the end

illwieckz replied:
Good catch! Fixed.
